### PR TITLE
Add calculator for number with arbitrary precision

### PR DESCRIPTION
**Pull request by TimotheePouchon**
In this PR we introduce an implementation of `IAlgebraicNumberCalculator` relying on the numbers with arbitrary precision from [PeterO.Numbers](https://github.com/peteroupc/Numbers).

### CODE REVIEW

File: KineticConvolution.sln [hunk 1]
```diff
@@ -25,6 +25,10 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "build", "build", "{7527BC77
 		.github\workflows\build-deploy.yml = .github\workflows\build-deploy.yml
 	EndProjectSection
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Hilke.KineticConvolution.EFloatAlgebraicNumber", "src\Hilke.EFloatCalculator\Hilke.KineticConvolution.EFloatAlgebraicNumber.csproj", "{81BC3082-2522-400C-998D-C644803C5384}"
```
Comment (thomashilke) [inline KineticConvolution.sln]:
```suggestion
Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Hilke.KineticConvolution.EFloatAlgebraicNumber", "src\Hilke.KineticConvolution.EFloatAlgebraicNumber\Hilke.KineticConvolution.EFloatAlgebraicNumber.csproj", "{81BC3082-2522-400C-998D-C644803C5384}"

```
The folder of the project is not named correctly

File: src/Hilke.EFloatCalculator/EFloatAlgebraicNumberCalculator.cs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+        public EFloatAlgebraicNumberCalculator(EContext context, EFloat tolerance)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            _tolerance = tolerance ?? throw new ArgumentNullException(nameof(tolerance));
+
+            if (tolerance.IsNegative)
+            {
+                throw new ArgumentOutOfRangeException(
+                    message: "Tolerance must be strictly positive.",
```
Comment (thomashilke) [inline src/Hilke.EFloatCalculator/EFloatAlgebraicNumberCalculator.cs]: Strictly or not strictly? `.IsNegative` is `false` for 0, or not?

File: src/Hilke.EFloatCalculator/Hilke.KineticConvolution.EFloatAlgebraicNumber.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <Nullable>Enable</Nullable>
+
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+
+    <IncludeSymbols>true</IncludeSymbols>
+    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
+
+    <IsPackable>true</IsPackable>
+    <Description>Implementation of a calculator with arbitrary precision numbers.</Description>
```
Comment (thomashilke) [inline src/Hilke.EFloatCalculator/Hilke.KineticConvolution.EFloatAlgebraicNumber.csproj]:
```suggestion
    <Description>Implementation of a calculator with arbitrary precision arithmetic.</Description>
```

File: src/Hilke.EFloatCalculator/EFloatAlgebraicNumberCalculator.cs [hunk 4]
```diff
@@ -0,0 +1,131 @@
+using System;
```
Comment (thomashilke) [inline src/Hilke.EFloatCalculator/EFloatAlgebraicNumberCalculator.cs]: The folder is wrong, it should be called `Hilke.KineticConvolution.EFloatCalculator`

File: tests/Hilke.EFloatCalculator.Tests/EFloatAlgebraicNumberCalculatorTests.cs [hunk 5]
```diff
@@ -0,0 +1,234 @@
+﻿using System;
```
Comment (thomashilke) [inline tests/Hilke.EFloatCalculator.Tests/EFloatAlgebraicNumberCalculatorTests.cs]: The folder is wrong, it should be `Hilke.KineticConvolution.EFloatCalculator.Tests`

**Review by thomashilke — CHANGES_REQUESTED** [review]

**Comment by Gimly** [comment]
@thomashilke Je profite de ce PR ouvert pour (faire un petit coucou 😄 et) vous avertir que vous avez "Rollomatic" dans le début du namespace dans le fichier README.md ainsi que dans ce fichier https://github.com/thomashilke/KineticConvolution/blob/10170c822eb38af2f2232d95703f6599a06b2f94/samples/Hilke.KineticConvolution.ReadMeExamples/WorkingExample.cs.

**Comment by thomashilke** [comment]
@Gimly Ouups bien vu, merci ^^ Bien le bonjour a toi! Petit spoiler: toute l'infrastructure de nombres algebriques qu'on a mise en place commence a bien porter ses fruit! ;)

File: KineticConvolution.sln (hunk 1 above)
Comment (TimotheePouchon) [reply KineticConvolution.sln]: all done

File: src/Hilke.EFloatCalculator/EFloatAlgebraicNumberCalculator.cs (hunk 2 above)
Comment (TimotheePouchon) [reply src/Hilke.EFloatCalculator/EFloatAlgebraicNumberCalculator.cs]: It's not so clear in the doc :
`Gets a value indicating whether this object is negative, including negative zero.`
I explicitly excluded the zero case.

**Pull request closed without merging**